src/Lib/Filters/ElasticSearch/TermsFilter.php

```diff
@@ -28,4 +31,25 @@ public function getValues()
 
         return $data;
     }
+
+    /**
+     * @return array
+     */
+    public function getFacetedValues(): array
+    {
+        $data = [];
+        foreach ($this->facets as $key => $value) {
+            $data[$value] = [$this->getName() => [$key => $value]];
+        }
+
+        return $data;
+    }
+
+    /**
+     * @return mixed|string
+     */
+    public function getName()
+    {
+        return $this->name;
```

Comment:
Why is this function here? It should be at base class (ElasticSearchFilterElement) level